bf2fc6cc711aee1a0c2a/kas-fleet-manager
# feat: cluster deregistration

## Description

Implements the deregistration of an Enterprise cluster. The SyncSet is removed and the cluster is soft-deleted afterwards.

## Verification Steps

1. Create an OCM cluster
2. Run the KFM locally, but don't target this cluster
3. Register the cluster using the enterprise registration endpoint
4. Make sure that registration was successful and that the Observability resources have been created
5. Deregister the cluster by calling the deregistration endpoint
6. Make sure that the Observability resources are removed, the cluster only soft deleted in the database.
